File: 프로그래머스/lv2/42842. 카펫/카펫.java
class Solution {
    public int[] solution(int brown, int yellow) {
    	//가로>=세로 길이 고정!!!
        int[] answer = new int[2]; //가로, 세로
        
        int s=brown+ yellow;//전체 넓이,  a*b
        int k=(int)Math.floor(Math.sqrt(s));
        //a=(brown+4)/2-b
        int a=s,b;
        while(a>=k) {
        	if (s%a!=0) {
        		a--; continue;}
            b=s/a;
        	if(brown==(a+b)*2-4) {
        		answer[0]=a; answer[1]=b;
                break;
        	}
        	else a--;
        }
        return answer;
    }
}

File: 프로그래머스/2/42842. 카펫/카펫.py
import math
def solution(brown, yellow):
    answer = []
    # 가로 y, 세로 x라고 할 때 / 가로 >= 세로
    # brown = 2*y + 2*x -4
    # yellow = (x-2)*(y-2) = x*y -2*x -2*y +4
    xy = brown+yellow
    for dx in range(3, int(math.sqrt(xy))+1):
        if xy%dx==0:
            dy = xy//dx
            if (dx-2)*(dy-2) == yellow:
                answer.append(dy)
                answer.append(dx)
                break
    return answer